### PR TITLE
Fix615

### PR DESCRIPTION
**Pull request by grepsuzette**
These commits fix both bugs mentionned in #615 for me.

I think it is easier for the reviewer to first revise commit after commit.
The changes are simple but the files modified are long and complicated, it can be daunting if not considering the logic.

So two bugs:

1. When testing a `<file>_testfile` and there were some panics (because for example the file was not placed in `file/` folder, or because bad package name), **panics** were not shown (see output in #615)

2. tests that panic-ed this way were unexpectedly shown as PASSED

The tests would need a major overhaul, but I tried to operate chirurgically to facilitate review.
Still, I decided to create two small files, as there is nothing to lose in doing that (see the 2 commits in question).

## How was it tested
It was only tested only manually, using combinations of foo_test.gno and bar_filetest.gno, with and without panics or `t.Fail()`. There might be edge cases. I recommend someone more experimented with realms tests it too.

### CODE REVIEW

File: cmd/gnodev/test.go [hunk 1]
```diff
@@ -298,17 +292,8 @@ func gnoTestPkg(
 			if err != nil {
 				errs = multierr.Append(errs, err)
 				io.ErrPrintfln("--- FAIL: %s (%s)", testName, dstr)
-				if verbose {
-					stdouterr, err := closer()
```
Comment (grepsuzette) [inline cmd/gnodev/test.go:302]: this was causing problem (by definition when it reached here because of line 288-291 `closer` was nil, causing a segfault because calling a 0x0000 pointer, the reason it was not apparent is because it only was called for _testfile AND when verbose AND when there was some error).

File: cmd/gnodev/test.go [hunk 2]
```diff
@@ -298,17 +292,8 @@ func gnoTestPkg(
...
-					if err != nil {
-						panic(err)
-					}
```
Comment (grepsuzette) [inline cmd/gnodev/test.go:305]: The panic here is now printed from above in a the general case.
I don't think we need this code anymore.

**Comment by grepsuzette** [comment]
I see some CI tests fail. Will take a look later.

**Comment by grepsuzette** [comment]
Closing, understand better the concept of filetest now and will post a cleaner fix later today.
